bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/connection/Connection.java

```diff
@@ -1413,9 +1414,9 @@ private void executeSequenceCommandWithVolume(List<Device> devices, @Nullable St
 
         JsonArray standardVolumeNodesToExecute = new JsonArray();
         for (int i = 0; i < devices.size(); i++) {
-            Integer ttsVolume = ttsVolumes.size() > i ? ttsVolumes.get(i) : null;
             Integer standardVolume = standardVolumes.size() > i ? standardVolumes.get(i) : null;
-            if (ttsVolume != null && standardVolume != null && !ttsVolume.equals(standardVolume)) {
+            Integer ttsVolume = ttsVolumes.size() > i ? ttsVolumes.get(i) : null;
+            if (ttsVolume != null && standardVolume != null && !standardVolume.equals(ttsVolume)) {
```

Comment:
```suggestion
            if (standardVolume != null && !standardVolume.equals(ttsVolume)) {
```

`.equals` is null-safe

Reply:
Yes but if we remove the null check of ttsVolume, it will reset the volume to standardVolume although it is the same as ttsVolume. For example standardVolume is 50 and ttsVolume is 50, it will not set the volume, but it would reset the volume. So we need to check if ttsVolume is set and if not, we also should not reset the volume.

Reply:
Because if we set or reset the volume and ttsVolume and standardVolume are equals, the Echo will still output a peep sound.

Reply:
The ttsVolume null check is for: "Do only reset the volume if we also set it"

Reply:
I just realized that when answering to your last comment....